src/annotation_processor.md

```diff
@@ -0,0 +1,149 @@
+---
+
+# Annotation Processor
+This sections introduces a generic schema annotation processing tool in Rest.Li. Before reading this, it is recommended to be familiar with the concept of PathSpec in Rest.Li. [Insert link here]
+
+## Annotating Rest.Li Schemas
+In Rest.Li, schema writers are able to add annotations to schema fields or the schema itself.
+
+**Example**: annotate on the schema fields
+```pdl
+record UserPersonallyIdentifiableInformation {
+  @data_classification = "MEDIUM"
+  firstName: string,
+  @data_classification = "MEDIUM"
+  lastName: string,
+  @data_classification = "LOW"
+  userId: long,
+  @data_classification = "HIGH"
+  socialSecurityNumber: string,
+}
+```
+
+**Example**: annotate directly on the schema
+```pdl
+@data_classification = "HIGH"
+record UserPersonallyIdentifiableInformation {
+  firstName: string,
+  lastName: string,
+  userId: long,
+  socialSecurityNumber: string,
+}
+```
+Note that these annotations are on the schema level.
+
+The support that Rest.Li provided for processing annotation in Rest.Li schema has been quite limited. These annotations in above examples are later stored as a `DataSchema`'s class attribute called "property". Just as the example shows, both field and the DataSchema can have this "property". We let the user chooses where they want to have the schema annotated and how they want to interpret. There were no preferrabled way regarding how annotation should be written, read and interpreted.
+
+## Inherit and override schema annotations
+But Rest.Li users found it useful to process annotations during schema processing. One use case is to introduce "inheritance" and "overrides" to annotations so those annotations can be dynamically processed in the way user defines, when for example the schemas were reused. This gives annotation extensibility and adds to schema reusability.
+
+Here are examples:
+Example case 1: Users might want the annotation of a field be inherited. The fields can reuse the annotations from upper level.
+```pdl
+@persistencePolicyInDays = 30
+record  UserVisitRecord {
+  visitedUrls: array[URL]
+  visitedUserProfiles: array[UserRecord]
+}
+```
+Reading from the schema, we might find both `visitedUrls` and `visitedUserProfiles` have `persistencePolicyInDays` annotated as 30.
+//<inheritance example>
+
+Example case 2: Users might want the annotation of a field be overriden. Override is where the annotation on a field or a schema might get updated when other annotations assign it another value.
+//<override example>
+```pdl
+@persistencePolicyInDays = 30
+record  UserVisitRecord {
+  @persistencePolicyInDays = 10
+  visitedUrls: array[URL]
+  visitedUserProfiles: array[UserRecord]
+}
+```
+Reading from the schema, a user might find the field `visitedUserProfiles`s `persistencePolicyInDays` annotation is 30 but `visitedUrls`'s `persistencePolicyInDays` annotation has value 10, which overrides the original value inherited(which was 30).
```

Comment:
some 's are not formatted correctly.

Reply:
I think they are. for example `visitedUrls` itself is one word. It is not plural.